test/sequential/test-repl-persistent-history.js

```diff
@@ -166,6 +172,18 @@ const tests = [{
   env: {},
   test: [UP],
   expected: [prompt, homedirErr, prompt, replDisabled, prompt]
+},
+{
+  env: { NODE_REPL_HISTORY: emptyHistoryPath,
+         NODE_REPL_HISTORY_FILE: emptyHistoryPath },
+  test: [UP],
+  expected: [prompt, sameHistoryFilePaths, prompt]
+},
+{
+  env: { NODE_REPL_HISTORY: defaultHistoryPath,
+         NODE_REPL_HISTORY_FILE: emptyHistoryPath },
+  test: [UP],
+  expected: [prompt, convertMsg, prompt]
```

Comment:
Ah I see, these still work since they do not use `os.homedir()`. Try removing the `NODE_REPL_HISTORY: defaultHistoryPath` from the last one, and you'll see why it should be moved, I think. :)